docs/pipeline.md

```diff
@@ -32,115 +33,171 @@ def build_pipeline():
-        node_pool_label="node_pool",  
-        node_pool_name="model-inference-pool",  
     )
     pipeline.add_op(caption_images_op, dependencies=load_from_hub_op)
     return pipeline
 
 ```
 
-In the example above, we first define our pipeline by providing a name as an identifier and a base path where the pipeline run artifacts will be stored. The base path can be a remote cloud location or a local directory, which is useful for local development.
+In the example above, we first define our pipeline by providing a name as an identifier and a base path where the pipeline run artifacts will be stored.
+
+The base path can be:
+
+* **A remote cloud location (S3, GCS, Azure Blob storage)**: valid across all runners. 
+For the **local runner**, make sure that your local credentials or service account have read/write access to the 
+designated base path and that they are mounted. <br>
+For the **Vertex** and **Kubeflow** runners, make sure that the service account attached to those runners has read/write access. 
+* **A local directory**: only valid for the local runner, points to a local directory. This is useful for local development.
 
 Next, we define two operations: `load_from_hub_op`, which is a based from a reusable component loaded from the Fondant registry, and `caption_images_op`, which is a custom component defined by you. We add these components to the pipeline using the `.add_op()` method and specify the dependencies between components to build the DAG.
 
 
 !!! note "IMPORTANT"
     Currently Fondant supports linear DAGs with single dependencies. Support for non-linear DAGs will be available in future releases.
 
-## Setting Custom node pool parameters
-Each component can optionally be constrained to run on particular node(s) using `node_pool_label` and `node_pool_name`. You can find these under the Kubernetes labels of your cluster. 
-You can use the default node label provided by Kubernetes or attach your own. Note that the value of these labels is cloud provider specific.  
-
-Note that you can also setup a component to use a preemptible VM by setting `preemptible` to `True`.
-This Requires the setup and assignment of a preemptible node pool. Note that preemptibles only work
-when KFP is setup on GCP. 
+## Compiling and Running a pipeline
 
-More info here: https://v1-6-branch.kubeflow.org/docs/distributions/gke/pipelines/preemptible/
+Once all your components are added to your pipeline you can use different compilers to run your pipeline:
 
+!!! note "IMPORTANT"
+  When using other runners you will need to make sure that your new environment has access to:
+  - The base path of your pipeline (as mentioned above)
+  - The images used in your pipeline (make sure you have access to the registries where the images are stored)
 
+### Local Runner
 
-## Setting Custom partitioning parameters
+This local runner is mainly aimed at local development and quick iterations, there is no scaling so using small slices of your data is advised.
```

Comment:
It does scale, right. Especially on a larger VM. I would mention that it only scales to the machine you're running on. And that switching to Vertex / KfP has advantages that you can choose hardware per component, get better monitoring, reproducibility, etc.